### PR TITLE
bpo-43452: Document the PyType_Lookup optimizations in the What's New for 3.10

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue43452

### CODE REVIEW

**Comment by pablogsal** [comment]
CC: @DinoV

**Comment by pablogsal** [comment]
Do we want to mention maybe some numbers from @methane's benchmark here: https://bugs.python.org/msg388485 ?

File: Doc/whatsnew/3.10.rst [hunk 1]
```diff
@@ -949,6 +948,10 @@ Optimizations
   algorithm to avoid quadratic behavior on long strings.  (Contributed
   by Dennis Sweeney in :issue:`41972`)
 
+* Added micro-optimizations to ``_PyType_Lookup()`` to improve cache lookup
```
Comment (vstinner) [inline Doc/whatsnew/3.10.rst]:
```suggestion
* Added micro-optimizations to ``_PyType_Lookup()`` to improve type attribute cache lookup
```

File: Doc/whatsnew/3.10.rst [hunk 2]
```diff
@@ -949,6 +948,10 @@ Optimizations
   algorithm to avoid quadratic behavior on long strings.  (Contributed
   by Dennis Sweeney in :issue:`41972`)
 
+* Added micro-optimizations to ``_PyType_Lookup()`` to improve type attribute cache lookup
+  performance in the common case of cache hits. (Contributed by Dino Viehland
+  in :issue:`43452`)
```
Comment (pablogsal) [inline Doc/whatsnew/3.10.rst]:
```suggestion
* Added micro-optimizations to ``_PyType_Lookup()`` to improve type attribute cache lookup
  performance in the common case of cache hits. This makes the interpreter 1.04 times faster
  in average (Contributed by Dino Viehland in :issue:`43452`)
```

**Pull request merged**